src/main/java/org/spongepowered/common/mixin/core/world/MixinWorld.java

```diff
@@ -2507,4 +2510,49 @@ public long getWeatherStartTime() {
     public void setWeatherStartTime(long weatherStartTime) {
         this.weatherStartTime = weatherStartTime;
     }
+
+    @Override
+    public EntityPlayer getClosestPlayerToEntityWhoAffectsSpawning(net.minecraft.entity.Entity entity, double distance) {
+        return this.getClosestPlayerWhoAffectsSpawning(entity.posX, entity.posY, entity.posZ, distance);
+    }
+
+    @Override
+    public EntityPlayer getClosestPlayerWhoAffectsSpawning(double x, double y, double z, double distance) {
+        double bestDistance = -1.0D;
+        EntityPlayer result = null;
+
+        for (Object entity : this.playerEntities) {
+            EntityPlayer player = (EntityPlayer) entity;
+            if (player == null || player.isDead || !((IMixinEntityPlayer) player).affectsSpawning()) {
+                continue;
+            }
+
+            double playerDistance = player.getDistanceSq(x, y, z);
+
+            if ((distance < 0.0D || playerDistance < distance * distance) && (bestDistance == -1.0D || playerDistance < bestDistance)) {
+                bestDistance = playerDistance;
+                result = player;
+            }
+        }
+
+        return result;
```

Comment:
IF this can return null, either mark it as `@Nullable` or make it an `Optional`.